### PR TITLE
Added aliases to custom navbar links and templates

### PR DESCRIPTION
**Pull request by adaybujeda**
Added aliases to static links and templates

Fixes [2460](https://github.com/OSC/ondemand/issues/2460)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1135148780858012/1203673227776906) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by johrstrom** [comment]
I feel like just extending the `Hash` class with an `alias_key` method or similar is nicer. 

```ruby
{ 
# ...
}.alias_key('logout', ['log out', 'log_out'])
```

With so few keys, I figure we can just duplicate the keys instead of trying to do something more complicated.  Maybe we could actually call it `duplicate_key` so that we don't get confused.

I tried to get something like this going, but couldn't.
```irb
hash = {["foo", "bar"]=>"foo"}
hash[hash.rassoc('foo').first] = 'baz'
=> {["foo", "bar"]=>"baz"}
```

**Comment by johrstrom** [comment]
> I was looking to extend the Hash class or other solutions, but they all seemed over-engineered compared to just setting the same value to multiple keys.

That sounds reasonable.

> What about renaming the utility function to:

That'll work for me.

**Comment by adaybujeda** [comment]
I have completed the changes to update the utility method name.

While waiting for the tests to go green, I thought about this solution, but I think I prefer the current utility method:

```ruby
    @@static_templates ||= MultikeyHash.new.tap do |hash|
      hash[['all_apps', 'all apps']] = 'layouts/nav/all_apps'
    end.freeze

  class MultikeyHash < Hash
    def []=(keys, value)
      keys.each do |key|
        super(key.to_sym, value)
      end
    end
  end
```

File: apps/dashboard/app/apps/nav_bar.rb [hunk 1]
```diff
@@ -166,4 +144,43 @@ def initialize(nav_item, partial_path)
...
   end
+
+  def self.set_multikey_value(hash: [], keys: [], value: nil)
+    keys.each do |key|
+      hash[key.to_sym] = value
+    end
+  end
+
+  def self.static_templates
+    @@static_templates ||= {}.tap do |hash|
```
Comment (johrstrom) [inline apps/dashboard/app/apps/nav_bar.rb:155]:
```suggestion
    @static_templates ||= {}.tap do |hash|
```

I think we want a class instance variable, not a class variable. Rubocop would tell you the same. 

https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/ClassVars

**Comment by johrstrom** [comment]
> While waiting for the tests to go green, I thought about this solution, but I think I prefer the current utility method:

If you're asking me, I think I prefer the class or core ext and freezing a constant. Though this'll work just the same - I can't tell if it's a question or just thinking out loud. For me, it's coming to what's the easiest to change or what would I see in another library.

**Comment by adaybujeda** [comment]
I have refactored to use a new inner class `MultikeyHash` and freezing the constants.
What do you think?

**Comment by johrstrom** [comment]
Yea this is fine by me.

**Comment by johrstrom** [comment]
I'm ready for this if you want to squash.

**Comment by adaybujeda** [comment]
Great!! Squash completed! - Ready to go 🚀

**Review by johrstrom — APPROVED** [review]

**Pull request merged**